shap/explainers/_deep/deep_pytorch.py

```diff
@@ -102,6 +97,7 @@ def remove_attributes(self, model):
                     pass
 
     def gradient(self, idx, inputs):
+        import torch
```

Comment:
We need the imports here because a DeepExplainer can be unpickled and the a call to gradient can be made, right?

Reply:
I just added the import here as `torch` functions are used further down in this function. I'm not aware of any pickle-related reasons for the torch import, but please do correct me if I'm wrong.

Reply:
No, the thing is, that I suspected that doing an import in the __init__ would be enough (ok, linters will complain). But then we might run into the pickle errors

Reply:
If we removed the import from the gradient method, wouldn't this line fail as `torch` is not defined within the local scope?

https://github.com/shap/shap/blob/0b73b8a41d005f120d909220d865a490cb002311/shap/explainers/_deep/deep_pytorch.py#L109

Reply:
I suppose we're on a bit of a tangent, but for what it's worth I think even if `__init__` is called first, the import would still needed again in `gradient` for the function to work as the module needs to be bound to a variable called `torch` in the local scope.

In any case I'll merge and we can discuss later, as I may well be missing something.